packages/tool-cache/src/manifest.ts

```diff
@@ -0,0 +1,96 @@
+import * as semver from 'semver'
+import {debug} from '@actions/core'
+
+// needs to be require for core node modules to be mocked
+// eslint-disable-next-line @typescript-eslint/no-require-imports
+import os = require('os')
+
+/*
+NOTE: versions must be sorted descending by version in the manifest
+      this library short circuits on first semver spec match
+[
+  {
+    "version": "1.2.3",
+    "stable": true,
+    "release_url": "https://github.com/actions/sometool/releases/tag/1.2.3-20200402.6",
+    "files": [
+      {
+        "filename": "sometool-1.2.3-linux-x64.zip",
+        "arch": "x64",
+        "platform": "linux",
+        "download_url": "https://github.com/actions/sometool/releases/tag/1.2.3-20200402.6/sometool-1.2.3-linux-x64.zip"
+      },
+    ...
+    ]
+  },
+  ...
+]
+*/
+
+export interface IToolReleaseFile {
+  filename: string
+  // 'aix', 'darwin', 'freebsd', 'linux', 'openbsd',
+  // 'sunos', and 'win32'
+  // in some images built for hosted it can also be suffixed by version
+  // e.g. linux-1804
+  platform: string
+
+  // 'arm', 'arm64', 'ia32', 'mips', 'mipsel',
+  // 'ppc', 'ppc64', 's390', 's390x',
+  // 'x32', and 'x64'.
+  arch: string
+
+  download_url: string
+}
+
+export interface IToolRelease {
+  version: string
+  stable: boolean
+  release_url: string
+  files: IToolReleaseFile[]
+}
+
+export async function _findMatch(
+  versionSpec: string,
+  stable: boolean,
+  candidates: IToolRelease[]
+): Promise<IToolRelease | undefined> {
+  const archFilter = os.arch()
+  const platFilter = os.platform()
+
+  let result: IToolRelease | undefined
+  let match: IToolRelease | undefined
+
+  let files: IToolReleaseFile | undefined
```

Comment:
`asset` or `file` will be better names because it is the single file

Reply:
OK.

Reply:
done